laboratory/views/organizations.py

```diff
@@ -0,0 +1,122 @@
+
+@author: luisza
+'''
+from __future__ import unicode_literals
+
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404
+from django.urls.base import reverse_lazy
+from django.utils.decorators import method_decorator
+from django import forms
+from mptt.forms import TreeNodeChoiceField
+from django.utils.functional import lazy
+from django.db.models import Q
+from laboratory.models import LaboratoryRoom, Laboratory, OrganizationStructure, PrincipalTechnician
+from laboratory.decorators import check_lab_permissions, user_lab_perms
+
+from .djgeneric import  ListView
+
+
+
+
+
+class OrganizationSelectableForm(forms.Form):
+    organizations = OrganizationStructure.objects.none()
+    filter_organization= TreeNodeChoiceField(queryset=organizations)
+
+    
+    def __init__(self,user, *args, **kwargs):
+        self.user = user 
+        super(OrganizationSelectableForm, self).__init__(*args, **kwargs)  
+        organizations = OrganizationStructure.objects.filter(
+            principaltechnician__credentials=self.user)
+        
+        orgs = None
+        for org in organizations:
+            if orgs is None:
+                orgs = Q (pk__in=org.get_descendants(include_self=True))
+            else:
+                orgs |= Q (pk__in=org.get_descendants(include_self=True) )   
+        
+        
+        if organizations.exists():               
+             self.fields['filter_organization'].queryset = OrganizationStructure.objects.filter(orgs).distinct()
+        elif self.user.is_superuser:     
+            self.fields['filter_organization'].queryset = OrganizationStructure.objects.all()
+    
+              
+@method_decorator(check_lab_permissions, name='dispatch')
+@method_decorator(login_required, name='dispatch')
+@method_decorator(user_lab_perms(perm='report'), name='dispatch')
+class OrganizationReportView(ListView):
+    model = Laboratory
+    template_name = "laboratory/report_organizationlaboratory_list.html"
+    organization = None
+
+    def get_context_data(self, **kwargs):
+        context = super(OrganizationReportView,self).get_context_data(**kwargs)
+        
+        context['form']  = self.form 
+        
+        # when a organizations is selected
+        if self.organization :
+            organizations_child = OrganizationStructure.objects.get(
+                pk=self.organization.pk
+                ).get_descendants(include_self=True)
+            labs=Laboratory.objects.filter(organization__in=organizations_child)
+            context['object_list'] = labs
+            context['filter_organization']  = self.organization
+            
+        # start report checking  technician  
+        elif self.technician :   
+            organizations = OrganizationStructure.objects.filter(
+                principaltechnician__credentials=self.user)
+            
+            orgs = None
+            for org in organizations:
+                if orgs is None:
```

Comment:
This code is repeated 3 times, so it's better if it's in a common utility function.  
If see it's the way to filter the user's organizations, so would be a common task.